src/NotepadNext/ScintillaNext.h

```diff
@@ -27,7 +27,7 @@
 #include <QFileInfo>
 
 
-static bool isRangeValid(const Sci_CharacterRange &range)
+[[maybe_unused]] static bool isRangeValid(const Sci_CharacterRange &range)
```

Comment:
Would keeping the definition in the header and then moving the implementation to the cpp file fix this warning? Because it is definitely used.

Reply:
but then it would not be inline. Do you need this to be inline?

Reply:
No. There's no known reason why this would be required to be inlined.

Reply:
This is a very small function, inlining it seems reasonable (creating a stack, and a call for such a trivial function seems "lame"). I have another approach. See the new commit.

Reply:
The static method is fine with me. Honestly, I wouldn't be surprised if the compiler optimizes this into an inline call anyways.

Reply:
This is why I kept it on the header, if it was on another compilation unit, the compiler will not be able to optimize it :)